crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -404,23 +411,25 @@ pub unsafe trait FetchState: Send + Sync + Sized {
 /// # Safety
 ///
 /// This must only be implemented for read-only fetches.
-pub unsafe trait ReadOnlyFetch {}
+pub unsafe trait ReadOnlyFetch<'w>: Fetch<'w> {}
```

Comment:
```suggestion
pub unsafe trait ReadOnlyFetch<'world>: Fetch<'world> {}
```

This should be consistent with `Fetch` above.